manim/mobject/types/vectorized_mobject.py

```diff
@@ -929,6 +929,32 @@ def get_nth_curve_function_with_length(
 
         length = np.sum(norms)
 
+        return length
+
+    def get_nth_curve_function_with_length(
+        self, n: int, sample_points: Optional[int] = None
+    ) -> typing.Tuple[typing.Callable[[float], np.ndarray], float]:
+        """Returns the expression of the nth curve along with its (approximate) length.
+
+        Parameters
+        ----------
+        n
+            The index of the desired curve.
+        sample_points
+            The number of points to sample to find the length.
+
+        Returns
+        -------
+        curve : typing.Callable[[float], np.ndarray]
+            The function for the nth curve.
+        length : :class:`float`
+            The length of the nth curve.
+        """
+
+        curve = self.get_nth_curve_function(n)
+
+        length = self.get_nth_curve_length(n, sample_points)
```

Comment:
This is less efficient than the previous implementation because to get the length of the curve you need to get the function anyways, which is the reason why I joined the returns, because `point_from_proportion` needed both.

Reply:
I made `get_nth_curve_length()` use `get_nth_curve_function_with_length`, so this should be fixed now.